src/box/relay.cc

```diff
@@ -1099,8 +1099,12 @@ tx_raft_msg_return(struct cmsg *base)
 {
 	struct relay_raft_msg *msg = (struct relay_raft_msg *)base;
 	msg->relay->tx.is_raft_push_sent = false;
+
 	if (msg->relay->tx.is_raft_push_pending)
```

Comment:
Why don't you run the triggers when there's a new broadcast? It might be not related to term anyhow. Even if it is related to term (e.g. another term bump), the logic in `box_raft_wait_term_broadcasted` will compare `sent_term` with `volatile_term` and everything will work correctly.

The trigger should run on every successful message despatch.

Reply:
Never mind. I misread this code chunk